File: leetcode/solutions/0387.FirstUniqueCharacterInAString/firstUniqChar_test.go
package solution

import "testing"

func Test_0387_FirstUniqueCharacterInAString(t *testing.T) {
	t.Parallel()

	tests := []struct {
		in  string
		out int
	}{
		{
			"leetcode",
			0,
		},
		{
			"loveleetcode",
			2,
		},
	}

	for _, tt := range tests {
		tt := tt

		t.Run("Test", func(t *testing.T) {
			t.Parallel()

			got := firstUniqChar(tt.in)

			if got != tt.out {
				t.Errorf("got %#v want %#v", got, tt.out)
			}
		})
	}
}

File: leetcode/solutions/0387.FirstUniqueCharacterInAString/firstUniqChar_test.ts
import { assertEquals } from "https://deno.land/std/testing/asserts.ts";
import firstUniqChar from "./firstUniqChar.ts";

Deno.test("0387. First Unique Character in a String", () => {
  type test = [string, number];
  const tests: test[] = [
    ["leetcode", 0],
    ["loveleetcode", 2],
  ];

  for (const [arg, want] of tests) {
    assertEquals(firstUniqChar(arg), want);
  }
});
